### PR TITLE
Add convert methods from and to BH.oM.DeepLearning.*

### LINKED ISSUE #5: Add conversion method from and to BHoM
#### Description:

Start porting some of the agnostic concepts in the BHoM to keras, like `Convolution2d`

### PR DESCRIPTION
**Pull request by epignatelli**
Depends on: https://github.com/BHoM/Keras.NET/pull/4

### Issues addressed by this PR

Closes #5

### Test files

https://burohappold.sharepoint.com/:u:/s/BHoM/EZcftkOdmGNDq3t2ufrKdAkBUtWsVHi5hl7R9FppHKwmgw?e=ccI8sx

### Changelog

- Added converts from Keras and from BHoM for:
  - `BH.oM.DeepLearning.IShape` -> `Keras.Shape`
  - `BH.oM.DeepLearning.LeakyReLU` -> `Keras.Layers.LeakyReLU`
  - `BH.oM.DeepLearning.ReLU` -> `Keras.Layers.ReLU`
  - `BH.oM.DeepLearning.Softmax` -> `Keras.Layers.Softmax`
  - `BH.oM.DeepLearning.AvgPooling2d` -> `Keras.Layers.AveragePooling2d`
  - `BH.oM.DeepLearning.Conv2d` -> `Keras.Layers.Conv2D`
  - `BH.oM.DeepLearning.GRU` -> `Keras.Layers.GRU`
  - `BH.oM.DeepLearning.LSTM` -> `Keras.Layers.LSTM`
  - `BH.oM.DeepLearning.MaxPooling2d` -> `Keras.Layers.MaxPooling2D`
  - `BH.oM.DeepLearning.TransposeConvolution2d` -> `Keras.Layers.Conv2DTranspose`
- Replace submodule dependency with NuGet
- Fix framework version to use .NET Framework 4.6.2

### CODE REVIEW

**Comment by epignatelli** [comment]
Hi guys,
This one is finally ready to be reviewed, would you mind checking if it makes sense?

No need deep checks as I will iterate on the toolkit. The converts will make sense when we can actually use the objects to perform a task.

**Comment by alelom** [comment]
I'm getting this:
image

**Comment by epignatelli** [comment]
Alright, that's an issue with Keras.NET. I'll push a fix for this.

**Comment by epignatelli** [comment]
Thanks @alelom ! I pushed a fix. 

This pr now depends on https://github.com/BHoM/Keras.NET/pull/4

**Comment by epignatelli** [comment]
There still is 1 manual thing to do, unfortunately, which does not depend on this repo.
You need to delete the `C:\User\%username%\AppData\Local\python-3.7.3-embed-amd64\python._pth` file.

**Comment by alelom** [comment]
Works
image
